sdk/eventhubs/azure-messaging-eventhubs-checkpointstore-blob/inc/azure/messaging/eventhubs/checkpointstore_blob/dll_import_export.hpp

```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+/**
+ * @file
+ * @brief DLL export macro.
+ */
+
+// For explanation, see the comment in azure/core/dll_import_export.hpp
+
+#pragma once
+
+/**
+ * @def AZ_MESSAGING_EVENTHUBS_DLLEXPORT
+ * @brief Applies DLL export attribute, when applicable.
+ * @note See https://docs.microsoft.com/cpp/cpp/dllexport-dllimport?view=msvc-160.
+ */
+
+#if defined(AZ_MESSAGING_EVENTHUBS_DLL) || (0 /*@AZ_MESSAGING_EVENTHUBS_DLL_INSTALLED_AS_PACKAGE@*/)
+#define AZ_MESSAGING_EVENTHUBS_BUILT_AS_DLL 1
+#else
+#define AZ_MESSAGING_EVENTHUBS_BUILT_AS_DLL 0
+#endif
+
+#if AZ_MESSAGING_EVENTHUBS_BUILT_AS_DLL
+#if defined(_MSC_VER)
+#if defined(AZ_MESSAGING_EVENTHUBS_BEING_BUILT)
+#define AZ_MESSAGING_EVENTHUBS_DLLEXPORT __declspec(dllexport)
```

Comment:
`AZ_MESSAGING_EVENTHUBS` here needs to be updated as well

Reply:
hmm, you are using the same inc structure as the eventhubs, I am not sure how is there no collision for `dll_import_export` and `rtti.hpp`

Reply:
The reason there's no collision is that the `checkpointstore_blob` directory doesn't exist in eventhubs.  

When you install this package, it overlays a `checkpointstore_blob` directory over the eventhubs inc tree.

That way you can say `#include <azure/messaging/eventhubs/checkpointstore_blob/blob_checkpoint_store.hpp>` to access the blob store.